frontend/apps/reader/modules/readergesture.lua

```diff
@@ -1479,9 +1479,8 @@ function ReaderGesture:gestureAction(action, ges)
         self.ui:handleEvent(Event:new("RestoreZoomMode"))
         self.ui:handleEvent(Event:new("InitScrollPageStates"))
     elseif action == "toggle_rotation" then
-        local event_name = self.document.info.has_pages and "SwapScreenMode" or "ChangeScreenMode"
-        local arg = Screen:getScreenMode() == "portrait" and "landscape" or "portrait"
-        self.ui:handleEvent(Event:new(event_name, arg))
+        local arg = (Screen:getRotationMode() +1)%4
```

Comment:
I generally do (+1 & 3) (as do the odd/even checks).

Reply:
Another strike against the modulo approach that the latest comment about -90° made me remember:

```
┌─(niluje@illyria:pts/9)─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(~/MPLAYER/FBInk)─┐
└─(0.65:27%:18:31:%)── echo $(( (0 - 1) & 3 ))                                                                                                                                                                                                                                                                                                         ──(Mon, Jun 29)─┘
3

┌─(niluje@illyria:pts/9)─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(~/MPLAYER/FBInk)─┐
└─(0.31:26%:18:37:%)── echo $(( (0 - 1) % 4 ))                                                                                                                                                                                                                                                                                                         ──(Mon, Jun 29)─┘
-1
```

;).

Reply:
Granted, it happens to behaves in Lua ;)

Reply:
needs bit module no builtin &